tests/test_mesh.py

```diff
@@ -873,18 +873,107 @@ def test_nativetag_expand():
     m.clam_001 = NativeMeshTag(1, float)
     assert_array_equal(m.clam_001[:], 2.2)
 
-def test_nativetag_mult():
+def test_nativetag_operator_setup():
     m = Mesh(structured=True, structured_coords=[[-1, 0, 1], [0, 1], [0,1]])
+    
     m.peach = NativeMeshTag(1, float)
     m.peach[:] = [1.5, 2.5]
-    m.tangerine = NativeMeshTag(1, float) 
+    
+    m.tangerine = NativeMeshTag(1, float)
     m.tangerine[:] = [5.0, 10.0]
+    
+    m.plum = NativeMeshTag(2, float)
+    m.plum[:] = [[5.0, 10.0], [15.0, 20.0]]
+
+    m.grapefruit = NativeMeshTag(2, float)
+    m.grapefruit[:] = [[1.0, 2.0], [4.0, 8.0]]
+    
+    m.quince = NativeMeshTag(3, float)
+    m.quince[:] = [[1.0, 2.0, 4.0], [5.0, 6.0, 8.0]]
+
+    return m
+
+def test_nativetag_add():
+    m = test_nativetag_operator_setup()
+
+    obs = m.peach + m.tangerine
+    exp = [6.5, 12.5]
+    assert_array_almost_equal(obs, exp)
+
+    obs = m.plum + m.grapefruit
+    exp = [[6.0, 12.0], [19.0, 28.0]]
+    assert_array_almost_equal(obs, exp)
+
+    cherry = 12.0
+    obs = m.peach + cherry
+    exp = [13.5, 14.5]
+    assert_array_almost_equal(obs, exp)
+
+    cherry = 5
+    obs = m.plum + cherry
+    exp = [[10.0, 15.0], [20.0, 25.0]]
+    assert_array_almost_equal(obs, exp)
+
+    cherry = [10.0, 12.0]
+    obs = m.peach + cherry
+    exp = [11.5, 14.5]
+    assert_array_almost_equal(obs, exp)
+
+    cherry = np.asarray(cherry)
+    obs = m.tangerine + cherry
+    exp = [15.0, 22.0]
+    assert_array_almost_equal(obs, exp)
+
+    cherry = "twelve"
+    assert_raises(TypeError, m.plum.__add__, cherry) #using invalid addend type
+
+    assert_raises(ValueError, m.quince.__add__, m.grapefruit) #using improper shape
+
```

Comment:
I have a hunch that `m.plum + m.tangerine` should work, by adding `m.tangerine` to each row of `m.plum`. Does it?

Reply:
Yes it does! Added tests for this.